packages/js/src/classic-editor/helpers/dom.js

```diff
@@ -5,173 +5,352 @@ import { excerptFromContent } from "../../helpers/replacementVariableHelpers";
+const createSetDomElementProp = ( domId, prop = "value" ) => ( value ) => set( document.getElementById( domId ), prop, value );
 
 /**
- * Gets the date month from the document.
+ * Gets the post title from the document.
  *
- * @returns {string} The date month or an empty string.
+ * @returns {string} The post title or an empty string.
  */
-export const getDateMonth = () => get( document.getElementById( DOM_IDS.DATE_MONTH ), "value", "" );
+export const getPostTitle = createGetDomElementProp( DOM_IDS.POST_TITLE );
 
 /**
- * Gets the date day from the document.
+ * Gets the term name from the document.
  *
- * @returns {string} The date day or an empty string.
+ * @returns {string} The term name or an empty string.
  */
-export const getDateDay = () => get( document.getElementById( DOM_IDS.DATE_DAY ), "value", "" );
+export const getTermName = createGetDomElementProp( DOM_IDS.TERM_NAME );
 
 /**
- * Gets the date year from the document.
+ * Gets the post content from the document.
  *
- * @returns {string} The date year or an empty string.
+ * @returns {string} The post content or an empty string.
  */
-export const getDateYear = () => get( document.getElementById( DOM_IDS.DATE_YEAR ), "value", "" );
+export const getPostContent = () => getContentTinyMce( DOM_IDS.POST_CONTENT ) || "";
 
 /**
- * Gets the date from the document.
+ * Gets the term description from the document.
  *
- * @returns {string} The date or an empty string.
+ * @returns {string} The term description or an empty string.
  */
-export const getDate = () => `${ getDateYear() }-${ getDateMonth() }-${ getDateDay() }`;
+export const getTermDescription = () => getContentTinyMce( DOM_IDS.TERM_DESCRIPTION ) || "";
 
 /**
- * Gets the meta description from the document.
+ * Gets the post date month from the document.
  *
- * @returns {string} The meta description or an empty string.
+ * @returns {string} The post date month or an empty string.
  */
-export const getMetaDescription = () => get( document.getElementById( DOM_YOAST_IDS.META_DESCRIPTION ), "value", "" );
+export const getPostDateMonth = createGetDomElementProp( DOM_IDS.POST_DATE_MONTH );
 
 /**
- * Gets the SEO title from the document.
+ * Gets the post date day from the document.
  *
- * @returns {string} The SEO title or an empty string.
+ * @returns {string} The post date day or an empty string.
  */
-export const getSeoTitle = () => get( document.getElementById( DOM_YOAST_IDS.SEO_TITLE ), "value", "" );
+export const getPostDateDay = createGetDomElementProp( DOM_IDS.POST_DATE_DAY );
 
+/**
+ * Gets the post date year from the document.
+ *
+ * @returns {string} The post date year or an empty string.
+ */
+export const getPostDateYear = createGetDomElementProp( DOM_IDS.POST_DATE_YEAR );
+
+/**
+ * Gets the post date from the document.
+ *
+ * @returns {string} The post date or an empty string.
+ */
+export const getPostDate = () => `${ getPostDateYear() }-${ getPostDateMonth() }-${ getPostDateDay() }`;
```

Comment:
3 remarks here:

1. If you really want to support the `or empty string` part you should change the implementation to something like this:

```suggestion
export const getPostDate = () => [ getPostDateYear(), getPostDateMonth(), getPostDateDay() ].filter( Boolean ).join( "-" );
```

But does not make much sense here. Maybe a `0` default per part instead of empty string for each part would make more sense? E.g. `0-0-0`. Or maybe something always valid, like `1` and `1970` or something?

2. Should this always be a valid date string or not? E.g. currently when you enter letters, it just pushes that as the date. I'm not very convinced that is a problem we should fix, but still.

3. WordPress has `OK` and `Cancel` in between these. It would be nice to support these. It is actually a bug now since the cancel is not picked up correctly.